### PR TITLE
add regexp to zip code input; tie labels to inputs

### PR DESCRIPTION
**Pull request by ericcf**
[#109445918]

### CODE REVIEW

**Comment by ericcf** [comment]
@eschlange nothing too exciting going on here. What's nice is that Angular handles client side validation for string length, regular expressions, etc. You just have to add the attribute, e.g. `minlength="5"` or `pattern="xyz"`.

File: public/partials/question.html [hunk 1]
```diff
@@ -15,6 +15,7 @@
       type="email"
       ng-required="question.required"
       name="{{ question.id }}"
+      id="{{ question.id }}"
```
Comment (eschlange) [inline public/partials/question.html]: Are the input names supposed to be identical to the input IDs?

**Comment by eschlange** [comment]
Looks good, just one question about the IDs and Names for input fields. Its good to merge otherwise.

File: public/partials/question.html (hunk 1 above)
Comment (ericcf) [inline public/partials/question.html]: In this case, it was the simplest thing for me to do. The names map onto the data being sent back to the server, and the ids map onto the associated label tags, so they serve different purposes.

Comment (eschlange) [inline public/partials/question.html]: Cool, just making sure it wasn't necessary to differentiate them.

**Pull request merged**